### DISCUSSION TITLE
Apps targetting children does require UMP GDPR Consent Form

### DISCUSSION THREAD

**Discussion by KiddiePlay** [airsdk/Adobe-Runtime-Support#2932] [General]
Hi Everyone
I have a problem and I don't know what to do
I use Animate CC 2024,,, And Air SDK 50.2.3.8 And Distriqt Anes
My application is directed to target children and I put the codes on this page :

[https://support.google.com/admob/answer/6223431?hl=en](https://support.google.com/admob/answer/6223431?hl=en&sjid=11912332898664718326-AP)

Which shows that the application is directed to children, especially ads, and so far there is no problem, and I have updated the application on the store....

Four days ago, Google sent me a warning about this topic (GDPR message) in my Admob account
Here is the problem:
image

And it tells me that the required codes regarding this matter must be placed in the application so that a message appears to users in this area (EEA and UK)...

I put the codes for this topic, but I stopped uploading the update because a friend told me that this library should not be put in the children's application, or in a more correct sense, that there is a conflict between the codes that are found here on this page
https://support.google.com/admob/answer/6223431?hl=en

And between the codes for this topic ( GDPR message and implementing the [UMP SDK]
(https://support.google.com/admob/answer/10113005))

UMP uses an AD ID which you are not allowed to use for apps targeting children.
And I put this library in my app (UMP) which uses the ads identifier to run successfully. The application was rejected by Google. I don't know whether to laugh or cry at this nonsense. Google is asking me to put a library (UMP) in the application, and when I put it, the application is rejected. What should I do now, and my application is aimed at children, not adults, and I know that it is not good to put the ads identifier in the application, especially children. But what do I do and this library(UMP) is asking for the ad ID. How is the problem solved. The topic has become very bad and causes me psychological problems

![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/95431188/605a8d24-e335-4c9b-a1f7-bec1df083860)
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/95431188/ebe82655-08f2-4158-9b99-0f292323f85f)

My Target Audience on Playstore its from: The target age group for your app is: Aged 5 and under, 6-8, 9-12, 13-15, 16-17, 18 and over
.. So Please Help me To Find a Solution.. Really Am getting confused and I cant Able to concentrate on any other work............ Pleasee Help me to find solution...............

**Comment by ajwfrost** [comment]
Hi

> Four days ago, Google sent me a warning about this topic (GDPR message) in my Admob account

I don't see the warning that they sent, but I would suggest that you were probably on the right track initially, not including this other library that requires the ad ID.

To comply with GDPR, there's no requirement to use a particular library/mechanism, you just need to disclose the uses of data and comply with the regulation. So it may be you can get away without using that UMP SDK?

Or alternatively if you set up Admob so that it's excluding the ad id completely, then maybe UMP needs configuring to say that it's not requiring access to this..?

I'm not an expert in this area though, and probably this question is better being posted on Google's/Admob forums?

thanks

**Reply by KiddiePlay** [reply]
Thank You So Much For your Response.. 
Anyways Am Still under confusion. What Changes should I need to make on Console under Data Safety Section ?

**Comment by pol2095** [comment]
see [here](https://stackoverflow.com/questions/71199143/action-required-your-app-is-not-compliant-with-google-play-policies-what-is-t/71234298#71234298) for "Issue found: Invalid Data safety form"

**Reply by KiddiePlay** [reply]
Hi,,Thank You For this Information.. That Portion is Clear For me Now. Really Thanks
I have Only One doubt on Target Audience : 
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/95431188/8dcbff95-6872-4882-a7fe-22a6dbd00f26)
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/95431188/7f630474-feaa-4ff4-afad-0627f7e26fd2)
Under Personal and sensitive information Settings, Do I need to Choose (Yes or No)...
Am New To Console. For Safety resaons am Asking. Please Can you clear me... Becasue Am Doing Kids Game.

**Reply by pol2095** [reply]
UMP has a parameter [setTagForUnderAgeOfConsent()](https://developers.google.com/admob/android/privacy?hl=en) by default `false`
```
// Set tag for under age of consent. false means users are not under age
    // of consent.
    ConsentRequestParameters params = new ConsentRequestParameters
        .Builder()
        .setTagForUnderAgeOfConsent(false)
        .build();
```
but for childrens must be `true`, to indicate that you want your content treated as child-directed for purposes of COPPA. This will prevent the transmission of the [Android advertising identifier (AAID)](https://support.google.com/googleplay/android-developer/answer/6048248).

I don't know if Google can detect this, ask Distriqt about this.

**Reply by pol2095** [reply]
I think, if you choose NO under Personal and sensitive information Settings, you need too to use in [AdMob](https://developers.google.com/admob/android/targeting?hl=en) :
```
RequestConfiguration requestConfiguration = MobileAds.getRequestConfiguration()
    .toBuilder()
    .setTagForChildDirectedTreatment(RequestConfiguration.TAG_FOR_CHILD_DIRECTED_TREATMENT_TRUE)
    .build();
MobileAds.setRequestConfiguration(requestConfiguration);
```
Ask Distriqt if it's implement this feature in their ANE.

**Reply by KiddiePlay** [reply]
Hi, 
Yes I Used this Code in My Ad Script.. Where ever We call Banner Ad, Interstitial Ad, Reward Ad. I Used this Code.

`private function getRequestConfiguration(): void {
			//log( "getRequestConfiguration" );
			var current: RequestConfiguration = Adverts.service.getRequestConfiguration();
			//log( current.toString() );
		}


		private function setRequestConfiguration(): void {
			//log( "setRequestConfiguration" );
			var requestConfiguration:RequestConfiguration = Adverts.service.getRequestConfiguration()
						.setTagForChildDirectedTreatment( RequestConfiguration.TAG_FOR_CHILD_DIRECTED_TREATMENT_TRUE )
						.setTagForUnderAgeOfConsent( RequestConfiguration.TAG_FOR_UNDER_AGE_OF_CONSENT_TRUE )
						.setMaxAdContentRating( RequestConfiguration.MAX_AD_CONTENT_RATING_G );
			
					Adverts.service.setRequestConfiguration( requestConfiguration );
		}`

So As I Understood I Need To Choose "NO" Under Personal And Sensitive Information Settings and 

Under Data Safety As you Said I need to Choose : "YES" For Device And ID Like In this Link : 

https://stackoverflow.com/questions/71199143/action-required-your-app-is-not-compliant-with-google-play-policies-what-is-t/71234298#71234298.

Am I Right Now..

**Reply by pol2095** [reply]
Submit changes to Play Store, wait and see...

**Reply by KiddiePlay** [reply]
Okay Sure, Is this Fine Right. because Am new to Console.. So Am little bit nervous

**Reply by pol2095** [reply]
Google Play becomes like Apple Store.

**Reply by KiddiePlay** [reply]
Okay,, Am Going to Test Today,,,, Thanks For you support. I will Let you know after feedback. 

Is apple Store easier or tough, when compared to google play store ? which is better for Game publisher..

**Reply by pol2095** [reply]
On iOS, there is also the App Tracking Transparency.

**Reply by KiddiePlay** [reply]
Okay, So Publish Games on Both Stores are Little Tough Now a days for Kid Games
